lib/socketpair.c

```diff
@@ -94,18 +96,31 @@ int Curl_socketpair(int domain, int type, int protocol,
     goto error;
   if(connect(socks[0], &a.addr, sizeof(a.inaddr)) == -1)
     goto error;
+
+  /* use non-blocking accept to make sure we don't block forever */
+  if(ioctlsocket(listener, FIONBIO, &nonblock) == -1)
+    goto error;
```

Comment:
You should probably use `curlx_nonblock()` instead for greater portability.

Reply:
I didn't see a point in using `curlx_nonblock` in Windows-only code.

Reply:
Oops, sorry, just realized my mistake, see https://github.com/curl/curl/pull/7144#discussion_r641521554